contentcuration/contentcuration/utils/recommendations.py

```diff
@@ -61,67 +68,200 @@ def create_backend(self) -> Backend:
+            return True
+        except Exception as e:
+            logging.exception(e)
+            return False
+
+    def get_recommendations(self, topic: Dict[str, Any],
+                            override_threshold=False) -> RecommendationsResponse:
+        """
+        Get recommendations for the given topic.
+
+        :param topic: A dictionary containing the topic for which to get recommendations. See
+        https://github.com/learningequality/le-utils/blob/main/spec/schema-embed_topics_request.json
+        :param override_threshold: A boolean flag to override the recommendation threshold.
+        :return: The recommendations for the given topic. :rtype: RecommendationsResponse
+        """
+        recommendations = []
+        request = EmbedTopicsRequest(
+            params={'override_threshold': override_threshold},
+            json=topic,
+        )
+        response = self.generate_embeddings(request=request)
+        nodes = self._extract_data(response)
+        if len(nodes) > 0:
+            node_ids = [node['contentnode_id'] for node in nodes]
+            recommendations = list(ContentNode.objects.filter(id__in=node_ids))
+
+        return RecommendationsResponse(results=recommendations)
+
+    def _extract_data(self, response: BackendResponse) -> List[Dict[str, Any]]:
+        return response.data if response.data is not None else []
+
+    def embed_content(self, nodes: List[ContentNode]) -> bool:
+        """
+        Embeds the content for the given nodes. This is an asynchronous process and could take a
+        while to complete. This process is handled by our curriculum automation service.
+        See https://github.com/learningequality/curriculum-automation
+
+        :param nodes: The nodes for which to embed the content.
+        :return: A boolean indicating that content embedding process has started.
+        :rtype: bool
+        """
+        for i in range(0, len(nodes), 20):
+            batch = nodes[i:i + 20]
+            content = [self.extract_content(node) for node in batch]
+            request = EmbedContentRequest(json=content)
+            self.generate_embeddings(request=request, cache=False)
+
+        return True
 
     def extract_content(self, node: ContentNode) -> Dict[str, Any]:
-        return {}
+        """
+        Extracts the content from the given node.
+
+        :param node: The node from which to extract the content.
+        :return: A dictionary containing the extracted content.
+        :rtype: Dict[str, Any]
+        """
+        contentkind_to_presets = {
+            content_kinds.AUDIO: [format_presets.AUDIO, format_presets.AUDIO_DEPENDENCY],
+            content_kinds.VIDEO: [
+                format_presets.VIDEO_DEPENDENCY,
+                format_presets.VIDEO_HIGH_RES,
+                format_presets.VIDEO_LOW_RES,
+                format_presets.VIDEO_SUBTITLE,
+            ],
+            content_kinds.EXERCISE: [format_presets.DOCUMENT, format_presets.EPUB],
```

Comment:
I doubt these will be associated with an exercise. Since we do have the `text` field in the request, and studio manages exercise questions and answers in the assessments, perhaps we should extract the assessment data and pass that through the `text` field.